src/main/java/org/terasology/projsndwv/genetics/Genome.java

```diff
@@ -18,14 +18,27 @@
 import org.terasology.projsndwv.genetics.components.GeneticsComponent;
 import org.terasology.utilities.random.MersenneRandom;
 
-import java.util.*;
-
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Handles genetic combination, including mutations.
```

Comment:
You're hitting a somewhat specialized field of study here where some users may know Java well and have some general ideas of typical gameplay mutations, but throwing in technical terms like locus and allele might make users wonder how that relates to gameplay logic. May be worth briefly covering at the class-level javadoc, maybe including a link to a wikipedia page covering the basic concepts?

Reply:
Whilst there is an argument that readers of the code should just be able to look things up, I think that having a link to the _specifc_ topic could help. It's also not reasonable to ask Soundwave to write up a full description if genetics in the javadoc. That's more relevant to the wiki.

Perhaps if we include some more detailed information in the wiki and refer people from the javadoc to either there or an external wiki which cab elaborate

Reply:
I'm hesitant to link to wikipedia articles about actual genetics, as while there is some assumption in the chosen nomenclature that analogies to actual genetics will be understood, the system makes no attempt to simulate actual genetics (especially not yet). This may be an appropriate place for a github wiki though.